### PR TITLE
Fix SIGINT/SIGTERM [V3]

### PR DESCRIPTION
**Pull request by apahim**
v3:
- Change the approach to fix this issue. Previously, I was trying to handle the signals from the main process all the way to the test sub-process, through all the layers in between. That was proven hand to control and with a big number of corner cases. Now, the approach is rather simple:
  - When a SIGINT in received by the main process, propagate the SIGINT for all the children.
  - When a SIGTERM is received by the main process, send a SIGKILL for all the children.
- Tests where rewritten to cover all the scenarios we have so far.

v2: #2137 
- Fix message on SIGINT (new commit)
- Improve SIGINT selftest.
- New commit to fix SIGINT/SIGTERM behavior on avocado.runner and process.Subprocess. See commit message for details.

v1: #2130

### CODE REVIEW

File: selftests/functional/test_interrupt.py [hunk 1]
```diff
@@ -67,24 +76,31 @@ def test_badly_behaved(self):
...
-                                          '(ignoring new Ctrl+C until then)')
-        # We have to actually wait 2 seconds until the ignore window is over
+        cmd = ('%s run %s --sysinfo=off --job-results-dir %s ' %
+               (AVOCADO, bad_test.path, self.tmpdir))
+        proc = subprocess.Popen(cmd.split(),
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+
+        def has_children():
+            return len(psutil.Process(proc.pid).children()) > 0
```
Comment (ldoktor) [inline selftests/functional/test_interrupt.py]: Well this can be just the `sysinfo` collection... Perhaps we can spawn 5 processes in test and wait for at least 6 children to be available...

File: selftests/functional/test_interrupt.py [hunk 2]
```diff
@@ -117,14 +133,20 @@ def wait_until_no_badtest():
 
             return len(bad_test_processes) == 0
 
-        wait.wait_for(wait_until_no_badtest, timeout=2)
+        if not wait.wait_for(wait_until_no_badtest, timeout=2):
+            self.fail('Avocado left processes behind.')
```
Comment (ldoktor) [inline selftests/functional/test_interrupt.py]: good catch

File: selftests/functional/test_interrupt.py [hunk 3]
```diff
@@ -67,24 +76,31 @@ def test_badly_behaved(self):
...
-        proc.sendline('\x03')
-        proc.read_until_last_line_matches('Interrupt requested. Waiting 2 '
-                                          'seconds for test to finish '
-                                          '(ignoring new Ctrl+C until then)')
-        # We have to actually wait 2 seconds until the ignore window is over
+        cmd = ('%s run %s --sysinfo=off --job-results-dir %s ' %
+               (AVOCADO, bad_test.path, self.tmpdir))
+        proc = subprocess.Popen(cmd.split(),
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
```
Comment (ldoktor) [inline selftests/functional/test_interrupt.py]: you only log output, it'd make sense to use `subprocess.STDOUT` for stderr.

File: selftests/functional/test_interrupt.py [hunk 4]
```diff
@@ -176,13 +208,159 @@ def wait_until_no_goodtest():
...
         # Make sure the Interrupted requested sentence is there
         self.assertIn('Interrupt requested. Waiting 2 seconds for test to '
-                      'finish (ignoring new Ctrl+C until then)',
-                      proc.get_output())
+                      'finish (ignoring new Ctrl+C until then)', output)
+
+    @unittest.skipIf(int(os.environ.get("AVOCADO_CHECK_LEVEL", 0)) < 2,
+                     "Skipping test that take a long time to run, are "
+                     "resource intensive or time sensitve")
+    def test_badly_behaved_sigterm(self):
```
Comment (ldoktor) [inline selftests/functional/test_interrupt.py]: This test can use most of the `test_badly_behaved_sigint` methods, this way it's really hard to follow what is different here and there. Could you please refactor it? (this way it'll become pain to maintain it)

File: selftests/functional/test_interrupt.py (hunk 1 above)
Comment (apahim) [inline selftests/functional/test_interrupt.py]: Or disable sysinfo?

File: selftests/functional/test_interrupt.py (hunk 3 above)
Comment (apahim) [inline selftests/functional/test_interrupt.py]: yep

File: selftests/functional/test_interrupt.py (hunk 4 above)
Comment (apahim) [inline selftests/functional/test_interrupt.py]: sure
